File: packages/sdk/output/typescript/src/services/passthrough/Passthrough.ts
import BaseService from '../../BaseService';

import { PassThroughResponse } from './models/PassThroughResponse';
import { PassThroughRequestDto } from './models/PassThroughRequestDto';

import { serializeQuery } from '../../http/QuerySerializer';

export class PassthroughService extends BaseService {
  async passthroughControllerPassthroughRequest(
    input: PassThroughRequestDto,
    integrationId: string,
    linkedUserId: string,
  ): Promise<PassThroughResponse> {
    if (integrationId === undefined || linkedUserId === undefined) {
      throw new Error(
        'The following are required parameters: integrationId,linkedUserId, cannot be empty or blank',
      );
    }
    const queryParams: string[] = [];
    const headers: { [key: string]: string } = { 'Content-type': 'application/json' };
    if (integrationId) {
      queryParams.push(serializeQuery('form', true, 'integrationId', integrationId));
    }
    if (linkedUserId) {
      queryParams.push(serializeQuery('form', true, 'linkedUserId', linkedUserId));
    }
    const urlEndpoint = '/passthrough';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}?${queryParams.join('&')}`);
    const response: any = await this.httpClient.post(
      finalUrl,
      input,
      {
        ...headers,
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data as PassThroughResponse;
    return responseModel;
  }
}

File: packages/sdk/output/python/src/testsdk/services/passthrough.py
from urllib.parse import quote
from ..net import query_serializer
from .base import BaseService
from ..models.PassThroughRequestDto import (
    PassThroughRequestDto as PassThroughRequestDtoModel,
)
from ..models.PassThroughResponse import PassThroughResponse as PassThroughResponseModel


class Passthrough(BaseService):
    def passthrough_controller_passthrough_request(
        self,
        request_input: PassThroughRequestDtoModel,
        linked_user_id: str,
        integration_id: str,
    ) -> PassThroughResponseModel:
        url_endpoint = "/passthrough"
        headers = {"Content-type": "application/json"}
        query_params = []
        self._add_required_headers(headers)
        if not integration_id:
            raise ValueError(
                "Parameter integration_id is required, cannot be empty or blank."
            )
        query_params.append(
            query_serializer.serialize_query(
                "form", False, "integrationId", integration_id
            )
        )
        if not linked_user_id:
            raise ValueError(
                "Parameter linked_user_id is required, cannot be empty or blank."
            )
        query_params.append(
            query_serializer.serialize_query(
                "form", False, "linkedUserId", linked_user_id
            )
        )
        final_url = self._url_prefix + url_endpoint + "?" + "&".join(query_params)
        res = self._http.post(final_url, headers, request_input, True)
        if res and isinstance(res, dict):
            return PassThroughResponseModel(**res)
        return res
